ui/src/dashboards/components/DashboardHeaderEdit.js

```diff
@@ -8,28 +8,46 @@ class DashboardEditHeader extends Component {
     const {dashboard: {name}} = props
     this.state = {name}
     this.handleChange = ::this.handleChange
+    this.handleFormSubmit = ::this.handleFormSubmit
+    this.handleKeyUp = ::this.handleKeyUp
   }
 
   handleChange(name) {
     this.setState({name})
   }
 
+  handleFormSubmit(e) {
+    e.preventDefault()
+    const name = e.target.name.value
+    this.props.onSave(name)
+  }
+
+  handleKeyUp(e) {
+    const {dashboard: {name}, onCancel} = this.props
+    if (e.key === 'Escape') {
+      this.setState({name})
```

Comment:
onCancel in `DashboardPage.js` does not do that, and it shouldn't. This is the proper place to do that. It has all the props information here.